kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -117,64 +114,79 @@ def execute(self, ask: AgentRequest) -> AnalyzerFixResponse:
 
         language = guess_language(ask.file_content, file_name)
 
+        source = " and ".join(ask.sources)
```

Comment:
I think its OK to put only one of the sources, it is not going to hurt the result a lot

Reply:
it's just sometimes the source is just `java` which I feel like could be misleading

Reply:
looks like we have got to add a new label on rules that we can use as a hint here, wdyt?

Reply:
Yeah maybe we can add a special label and if it's present grab that instead. Not sure how hard it would be up add but probably not that bad